### ISSUE TITLE
False-positive R0801:  Similar lines in 2 files

### ISSUE THREAD

**Issue by benhacka** [PyCQA/pylint#3239]
##### File with abstract class
```python
"""abs_class.py"""

from abc import ABC, abstractmethod


class AbcTest(ABC):

    @abstractmethod
    def do_something_else(self) -> bool:
        """Do something else here"""

    @property
    @abstractmethod
    def something(self) -> int:
        """Do something here"""

```

##### The first file with a class that inherits from abstract

```python
"""test_class1.py"""

from abs_class import AbcTest


class RealTest1(AbcTest):
    _a = 0

    def do_something_else(self) -> bool:
        ...
        self._a = 1 + 1
        return True

    @property
    def something(self) -> int:
        self.do_something_else()

        return self._a


```

##### The second file with a class that inherits from abstract

```python
"""test_class2.py"""

from abs_class import AbcTest


class RealTest2(AbcTest):
    _a = 0

    def do_something_else(self) -> bool:
        ...
        self._a = 1 - 1
        return True

    @property
    def something(self) -> int:
        self.do_something_else()

        return self._a


```

As you can see, these are different methods, with different logic.

### Current behavior
```
venv\__init__.py:1:0: R0801: Similar lines in 2 files
==test_class1:8
==test_class2:8
        return True

    @property
    def something(self) -> int:
        self.do_something_else()

        return self._a (duplicate-code)

```

### Expected behavior

Do not show a message about similar lines if the abstract property has been overridden and methods stay in the same order in several modules.

### pylint --version output
pylint 2.4.3
astroid 2.3.3
Python 3.7.4 (tags/v3.7.4:e09359112e, Jul  8 2019, 20:34:20) [MSC v.1916 64 bit (AMD64)]

**Comment by PCManticore** [comment]
Thanks for opening an issue. This check is completely semantic-less, it does not care about what the code represents, only how it looks. And to this check, these lines seem the same and thus the error you are getting. There are a couple of things you can do here:

- increase the line limit for `duplicate-code` with `min-similarity-lines=`

- don't mark `something` as an abstract property given it's the same for these two implementation classes and define it in the base class

Hope this helps. Unfortunately we won't be adding any semantic features to `duplicate-code` as it's out of its scope.
